src/tools/clippy/tests/ui/cast_ref_to_mut.stderr

```diff
@@ -1,22 +1,38 @@
+error: immediately dereferencing a reference
+  --> $DIR/cast_ref_to_mut.rs:20:9
+   |
+LL |         *&mut *(a as *const _ as *mut String) += " world";
+   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try this: `*(a as *const _ as *mut String)`
+   |
+   = note: `-D clippy::deref-addrof` implied by `-D warnings`
```

Comment:
Oof. I've missed this while committing, this is annoying.

So this lint sees `*ptr +=` and notices that there is an implicit auto-ref of a place derived from a pointer, so it suggests `&mut *ptr +=`. But this doesn't work! So, I add a `*` to make it work... which distresses clippy :|

I even think that `*&mut *` is the correct way to write this so uhhh... Should I `#[allow]` either this lint or `clippy::deref-addrof`?

Reply:
`*ptr = 5` still works without explicit `&mut`, I assume, so maybe `*ptr += 5` should also still work?

Reply:
`=` doesn't go through a reference though, `+=` is user code after all, it's literally `AddAssign::add_assign(&mut ..., ...)`, so I don't really see a difference between it and normal functions.

Reply:
Hm... in the common case, `+=` will actually write to its argument though, so the extra `&mut` makes no difference.

The cases in the two linked issues are different since those are functions that do not write to their argument reference at all, so there's extra (avoidable) UB from the reference.

Reply:
Well, we could maybe check that the outer expression is a call to `*Assign::*_assign` and ignore the lint to make it less noisy. That said though it would take more or less the same as better suggestions for `len` & co.

Reply:
So HIR does not have a primitive `+=`, this is already a method call?

Reply:
I was mistaken, HIR does have a `ExprKind::OpAssign`

Reply:
This thread is still open and the diff still shows. I think for now the lint should probably just not fire on `+=`, just like it does not fire on `=`?